airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/record.py

```diff
@@ -0,0 +1,119 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+from io import TextIOWrapper
+from json import loads
+from typing import Any, Iterable, List, Mapping, Optional, Union
+
+from .query import ShopifyBulkQuery
+from .tools import END_OF_FILE, BulkTools
+
+
+class ShopifyBulkRecord:
```

Comment:
can you add some unit tests for the methods in this class? Only record_resolve_id seems to be tested

Reply:
The `record.py` module is covered by 98%, I've added some edge case tests for `check_types`. Now it's 100%.

Reply:
the value I see in unit tests specific to this class is it'll document the expected shape of the records before and after calling each method.

as far as I can tell the coverage is due to `test_bulk_stream_parse_response`. that's a good integration test but it does little to help understand how this class is meant to work